inc/class-tmsc-custom-landing-page-types.php

```diff
@@ -0,0 +1,816 @@
+			}
+
+			if ( ! empty( $this->current_term->term_id ) ) {
+				$url = add_query_arg( array( $this->current_taxonomy => $this->current_term->term_id ), $url );
+			}
+		}
+		return $url;
+	}
+
+	/**
+	 * We hard override our status views with our custom set.
+	 */
+	public function get_current_views() {
+		$args = array(
+			'post_type' => $this->current_post_type,
+			'taxonomy' => $this->current_taxonomy,
+		);
+
+		$all_attrs = ( empty( $_GET[ $this->current_taxonomy ] ) && empty( $_GET['post_status'] ) && empty( $_GET['landing_type'] ) ) ? array( 'class' => 'current' ) : array();
+		$parent_attrs = ( ! empty( $_GET['landing_type'] ) && ! empty( $this->current_landing_type ) && 'archive' === $this->current_landing_type ) ? array( 'class' => 'current' ) : array();
+		$subpages_attrs = ( ! empty( $this->current_landing_type ) && 'single' === $this->current_landing_type ) ? array( 'class' => 'current' ) : array();
+		$trash_attrs = ( ! empty( $_GET['post_status'] ) && 'trash' === $_GET['post_status'] ) ? array( 'class' => 'current' ) : array();
+
+		return array(
+			'all' => $this->get_admin_link( 'edit.php', $args, __( 'All', 'ai' ), $all_attrs ),
+			'parents' => $this->get_admin_link( 'edit.php', array_merge( $args, array( 'landing_type' => 'archive' ) ), $this->types[ $this->current_taxonomy ]['labels']['name'], $parent_attrs ),
+			'subpages' => $this->get_admin_link( 'edit.php', array_merge( $args, array( 'landing_type' => 'single' ) ), $this->types[ $this->current_taxonomy ]['labels']['child_name'], $subpages_attrs ),
+			'trash' => $this->get_admin_link( 'edit.php', array_merge( $args, array( 'post_status' => 'trash' ) ), __( 'Trash', 'ai' ), $trash_attrs ),
+		);
+	}
+
+	/**
+	 * Set the urls for the status filter links on the edit pages.
+	 * @param array $views. Associative array of status keys with corresponding links.
+	 * @return array;
+	 */
+	public function set_status_filter_urls( $views ) {
+		if ( $this->on_custom_landing_admin_page() ) {
+			$this->current_views = $this->get_current_views();
+			return $this->current_views;
+		}
+		return $views;
+	}
+
+	/**
+	 * Add in an action to see subpages on parent page.
+	 * @param array $actions. array( 'class' => 'HTML' ).
+	 * @param object $post. WP Post Object.
+	 * @return array.
+	 */
+	public function add_view_subpage_link_action( $actions, $post ) {
+		if ( $this->on_custom_landing_admin_page( $post ) && $this->is_parent_landing_page( $post ) ) {
+			$term = tmsc_get_linked_term( $post->ID );
+
+			$link = '<a href="' . esc_url( add_query_arg( array( 'post_type' => $this->current_post_type, $this->current_taxonomy => $term->term_id ), 'edit.php' ) ) . '" aria-label="' . esc_html__( 'View ', 'ai' ) . esc_attr( $post->post_title ) . ' ' . esc_html( $this->types[ $this->current_taxonomy ]['labels']['child_name'] ) . '">' . esc_html( $this->types[ $this->current_taxonomy ]['labels']['child_name'] ) . '</a>';
```

Comment:
You have some `esc_html` functions that should be `esc_attr`.  Also, please update the text domain.